Docs: Add note about muslc incompatibility (#8162)
github/codeql: docs/codeql/codeql-cli/getting-started-with-the-codeql-cli.rst

**Information for macOS "Catalina" (or newer) users**. If you are using macOS
on Apple Silicon (e.g. Apple M1), ensure that the `Xcode command-line developer
tools <https://developer.apple.com/downloads/index.action>`__ and `Rosetta 2
<https://support.apple.com/en-us/HT211861>`__ are installed. 

> Please note that the CodeQL CLI is not compatible with non-glibc Linux 
> distributions such as (muslc-based) Alpine Linux at this time.

Reviewer: Put this into a note box:
```suggestion
.. pull-quote:: Note

   The CodeQL CLI is currently not compatible with non-glibc Linux 
   distributions such as (muslc-based) Alpine Linux.
```
